File: Creational/Method Chaining/java/MethodChaining.java
public class MethodChaining {
	public static void main(String[] args) {
		Rectangle rect1 = new Rectangle();
		Rectangle rect2 = new Rectangle();

		System.out.println(rect1);
		//This is a 0.0x0.0 Rectangle

		System.out.println(rect2.setWidth(5).setHeight(2));
		//This is a 5.0x2.0 Rectangle
	}

	public static class Rectangle {
		private static final double DEFAULT_WIDTH = 0;
		private static final double DEFAULT_HEIGHT = 0;

		private double width;
		private double height;

		public Rectangle() {
			height = DEFAULT_WIDTH;
			width = DEFAULT_HEIGHT;
		}

		public Rectangle setWidth(double width) {
			this.width = width;
			return this;
		}

		public Rectangle setHeight(double height) {
			this.height = height;
			return this;
		}


		public String toString() {
			return "This is a "+width+"x"+height+" Rectangle";
		}

	}
}

File: Creational/Method Chaining/php/MethodChaining.php
<?php

class Rectangle {
    public const DEFAULT_WIDTH = 0;
    public const DEFAULT_HEIGHT = 0;
    protected $width;
    protected $height;
    
    public function __construct(int $width = 0, int $height = 0)
    {
        $this->width = $width;
        $this->height = $height;
    }

    public function setWidth(int $width)
    {
        $this->width = $width;
        return $this;
    }

    public function setHeight(int $height)
    {
        $this->height = $height;
        return $this;
    }

    public function __toString()
    {
        return sprintf("\n\nThis is a %dx%d Rectangle\n\n", $this->width, $this->height);
    }
}

print(new Rectangle());
$rect = new Rectangle();
print($rect->setWidth(10)->setHeight(5));
